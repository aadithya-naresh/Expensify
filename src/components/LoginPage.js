import React from 'react'
import { connect } from 'react-redux'
import {startLogin} from '../actions/auth'

const LoginPage = ({ startLogin }) =>(
    <div>
        <h1>Login Page</h1>
        <button onClick={startLogin}>Login Button</button>
    </div>
)
const mapDispatchToProps = (dispatch) =>({
    startLogin : () => dispatch(startLogin())
})
export default connect(undefined,mapDispatchToProps)(LoginPage)