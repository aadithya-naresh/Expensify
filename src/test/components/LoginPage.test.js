import React from 'react'
import {shallow} from 'enzyme'
import LoginPage from '../../components/LoginPage'

test('should render login page', () =>{
    const wrapper = shallow(<LoginPage startLogin={() => { }}/>)
    expect(wrapper).toMatchSnapshot()
})

test('Should call startlogin on login button click', () =>{
    const startLogin= jest.fn()
    const wrappper = shallow(<LoginPage startLogin={startLogin}/>)
    wrappper.find('button').simulate('click')

    expect(startLogin).toHaveBeenCalled()
})