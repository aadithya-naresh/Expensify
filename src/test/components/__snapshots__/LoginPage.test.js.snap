// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render login page 1`] = `
<div>
  <h1>
    Login Page
  </h1>
  <button>
    Login Button
  </button>
</div>
`;
